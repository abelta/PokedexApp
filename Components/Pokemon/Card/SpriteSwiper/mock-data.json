{
  "sprites": {
      "back_female": null,
      "back_shiny_female": null,
      "back_default": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/6.png",
      "front_female": null,
      "front_shiny_female": null,
      "back_shiny": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/6.png",
      "front_default": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png",
      "front_shiny": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/6.png"
  }
}
