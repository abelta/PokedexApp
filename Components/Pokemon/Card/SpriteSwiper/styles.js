import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  spriteSwiper: {
    backgroundColor: 'red',
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
  },
  sprite: {
    height: 100,
    width: 100,
  },
});

export default styles;
