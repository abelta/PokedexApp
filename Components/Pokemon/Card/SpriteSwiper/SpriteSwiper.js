import React from 'react';
import PropTypes from 'prop-types';
import { Image } from 'react-native';
import Swiper from 'react-native-swiper';
import styles from './styles';

const SpriteSwiper = ({ autoplay, data }) => (
  <Swiper
    autoplay={autoplay}
    showsPagination={false}
    style={styles.spriteSwiper}
  >
    {
      data
      .filter(uri => uri)
      .map(uri => <Image key={uri} source={{ uri }} style={styles.sprite} />)
    }
  </Swiper>
);

SpriteSwiper.propTypes = {
  autoplay: PropTypes.bool,
  data: PropTypes.arrayOf(PropTypes.string).isRequired,
};

SpriteSwiper.defaultProps = {
  autoplay: true,
};

export default SpriteSwiper;
