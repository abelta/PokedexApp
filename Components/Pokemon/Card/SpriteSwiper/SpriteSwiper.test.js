import React from 'react';
import renderer from 'react-test-renderer';
import SpriteSwiper from '.';
import data from './mock-data.json';

describe('SpriteSwiper', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <SpriteSwiper autoplay={false} data={Object.values(data.sprites)} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
