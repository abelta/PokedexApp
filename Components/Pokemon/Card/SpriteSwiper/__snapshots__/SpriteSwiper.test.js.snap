// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpriteSwiper renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "position": "relative",
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    autoplay={false}
    autoplayDirection={true}
    autoplayTimeout={2.5}
    bounces={false}
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {},
      ]
    }
    contentOffset={
      Object {
        "x": 0,
      }
    }
    disableNextButton={false}
    horizontal={true}
    index={0}
    loadMinimal={false}
    loadMinimalSize={1}
    loop={true}
    onIndexChanged={[Function]}
    onMomentumScrollEnd={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={true}
    scrollsToTop={false}
    showsButtons={false}
    showsHorizontalScrollIndicator={false}
    showsPagination={false}
    showsVerticalScrollIndicator={false}
    style={Object {}}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/6.png",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 100,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/6.png",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 100,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/shiny/6.png",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 100,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 100,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/shiny/6.png",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 100,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "uri": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/6.png",
            }
          }
          style={
            Object {
              "height": 100,
              "width": 100,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
